shared/metrics/src/main/java/io/pravega/shared/metrics/StatsLoggerImpl.java

```diff
@@ -77,100 +79,102 @@ public StatsLogger createScopeLogger(String scope) {
         if (0 == basename.length()) {
             scopeName = scope;
         } else {
-            scopeName = name(basename, scope);
+            scopeName = basename + "." + scope;
         }
         return new StatsLoggerImpl(metrics, scopeName);
     }
 
     private class CounterImpl implements Counter {
-        private final com.codahale.metrics.Counter counter;
+        private io.micrometer.core.instrument.Counter counter;
+        private final io.micrometer.core.instrument.Tags tags;
         @Getter
+        private final Id id;
         private final String name;
 
-        CounterImpl(String statName) {
-            this.name = name(basename, statName);
-            this.counter = metrics.counter(name);
+        CounterImpl(String statName, String... tagPairs) {
+            this.tags = io.micrometer.core.instrument.Tags.of(tagPairs);
+            this.name = basename + "." + statName;
+            this.counter = metrics.counter(name, this.tags);
+            this.id = counter.getId();
         }
 
         @Override
-        public void close() {
-            metrics.remove(this.name);
-        }        
+        public synchronized void close() {
+            metrics.remove(counter);
+        }
 
         @Override
         public synchronized void clear() {
-            long cur = counter.getCount();
-            counter.dec(cur);
+            metrics.remove(counter.getId());
+            this.counter = metrics.counter(name, tags);
         }
 
         @Override
         public long get() {
-            return counter.getCount();
+            return (long) counter.count();
```

Comment:
This method needs to be `synchronized`

Reply:
fixed

Reply:
`@Synchronized` is not the same as adding the `synchronized` modifier to the method. Please be consistent.

In addition, mark the `counter` field as `@GuardedBy("this")` (assuming you go with the `synchronized` modifier).